src/psyclone/psyir/backend/fortran.py

```diff
@@ -496,11 +479,74 @@ def codeblock_node(self, node):
         PSyIR tree. It returns the content of the CodeBlock as a
         Fortran string, indenting as appropriate.
 
+        At the moment it is not possible to distinguish between a
+        codeblock that is one or more full lines (and therefore needs
+        a newline added) and a codeblock that is part of a line (and
+        therefore does not need a newline). The current implementation
+        adds a newline irrespective. This is the subject of issue
+        #388.
+
         :param node: A CodeBlock PSyIR node.
         :type node: :py:class:`psyclone.psyGen.CodeBlock`
 
         :returns: The Fortran code as a string.
         :rtype: str
 
         '''
-        return self._nindent.join(str(node.ast).splitlines(True))
+        result = ""
+        for statement in node.get_statements:
+            result += "{0}{1}\n".format(self._nindent, str(statement))
```

Comment:
I don't like the concatenations like this anywhere - adding strings is very inefficient in python (strings are unmutable). It's much faster to assemble the strings in a list, and then join them. I won't request a change, but leave it to a discuss during our next telco. But this just asks for:
```
result_list = []
for statement in node.get_statements:
   result_list.extend( [self.n_indent, str(statement)] )
result_list.append("")
return "\n".join(result_list)
```
OK, that code wasn't tested, but just to get the idea (maybe we can even return (self._nintend+"\n").join(...), if we set up the list correctly.

Reply:
OK, so we'll discuss at the next telco rather than chaning here.